src/datasets/info.py

```diff
@@ -189,13 +206,18 @@ def unique(values):
         features = None
         supervised_keys = None
 
+        # Find common task templates across all dataset infos
+        all_task_templates = [info.task_templates for info in dataset_infos]
+        task_templates = list(set(all_task_templates[0]).intersection(*all_task_templates[1:]))
```

Comment:
@SBrandeis this is a first pass at trying to find common members across task templates. 

since the task template classes are unhashable by default, i defined my own `__hash__` functions for `TextClassification` and `QuestionAnswering`. i'm not really happy with this approach (it feels clunky and prone to failure), so if you have a better idea on computing the intersection across task templates then i'm all ears!

Reply:
Using the hash is a good idea IMO if we are certain two TaskTemplates with the same attributes have the same hash!

I believe we can achieve that by decorating the classes with `@dataclass(frozen=True)`